sources/api/migration/migrator/src/main.rs

```diff
@@ -23,45 +23,88 @@
 #[macro_use]
 extern crate log;
 
+use args::Args;
+use error::Result;
 use nix::{dir::Dir, fcntl::OFlag, sys::stat::Mode, unistd::fsync};
 use rand::{distributions::Alphanumeric, thread_rng, Rng};
 use semver::Version;
 use simplelog::{Config as LogConfig, TermLogger, TerminalMode};
 use snafu::{ensure, OptionExt, ResultExt};
 use std::collections::HashSet;
 use std::env;
-use std::fs::{self, Permissions};
+use std::fs::{self, File, Permissions};
 use std::os::unix::fs::{symlink, PermissionsExt};
 use std::os::unix::io::AsRawFd;
 use std::path::{Path, PathBuf};
 use std::process::{self, Command};
+use tempfile::TempDir;
+use tough::ExpirationEnforcement;
+use update_metadata::{load_manifest, Direction, MIGRATION_FILENAME_RE, REPOSITORY_LIMITS};
+
+#[macro_use]
+extern crate lazy_static;
 
-use update_metadata::MIGRATION_FILENAME_RE;
 mod args;
-mod direction;
 mod error;
 
-use args::Args;
-use direction::Direction;
-use error::Result;
+lazy_static! {
+    /// This is the last version of Bottlerocket that supports *only* unsigned migrations.
+    static ref LAST_UNSIGNED_MIGRATIONS_VERSION: Version = Version::new(0, 3, 4);
+}
 
 // Returning a Result from main makes it print a Debug representation of the error, but with Snafu
 // we have nice Display representations of the error, so we wrap "main" (run) and print any error.
 // https://github.com/shepmaster/snafu/issues/110
 fn main() {
-    if let Err(e) = run() {
+    let args = Args::from_env(env::args());
+    // TerminalMode::Mixed will send errors to stderr and anything less to stdout.
+    if let Err(e) = TermLogger::init(args.log_level, LogConfig::default(), TerminalMode::Mixed) {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+    if let Err(e) = run(&args) {
         eprintln!("{}", e);
         process::exit(1);
     }
 }
 
-fn run() -> Result<()> {
-    let args = Args::from_env(env::args());
+#[deprecated(since = "0.3.5", note = "for unsigned migrations.")]
```

Comment:
```suggestion
```
It doesn't seem like this function is deprecated?  You'd still use it to determine whether migrations are signed after that version...

Reply:
Addressed in [e864eba](https://github.com/bottlerocket-os/bottlerocket/compare/acd95cfacb27e120daf9e903dd4b6cd0f2f8a015..e864eba052e35a165bd07ae9a04c1bb410daf6bd)